src/DocumentFragment.js

```diff
@@ -0,0 +1,38 @@
+  // ### DocumentFragment Shim
+  //
+  // A representation of a document fragment that can be normally generated
+  // by calling `document.createDocumentFragment`. Since the document fragment
+  // is itself a node, it inherits all properties of a node.
+module.exports = (function(){
+  "use strict";
+
+  var Node = require('./Node');
+
+  function DocumentFragment() {
+    Node.call(this, Node.DOCUMENT_FRAGMENT_NODE, "#document-fragment");
+  }
+  DocumentFragment.prototype = Object.create(Node.prototype);
+  DocumentFragment.prototype.constructor = DocumentFragment;
+
+  DocumentFragment.prototype.createTextNode = function(data) {
+    return new Node(Node.TEXT_NODE, "#text", data);
+  };
+
+  DocumentFragment.prototype.createElementNS = function(namespaceURI, qualifiedName) {
+    return new Node(Node.ELEMENT_NODE, qualifiedName, null, namespaceURI);
+  };
+
+  DocumentFragment.prototype.createElement = function(tagName) {
+    return this.createElementNS(null, tagName);
+  };
+
+  DocumentFragment.prototype.createComment = function(data) {
+    return new Node(Node.COMMENT_NODE, "#comment", data);
+  };
+
+  DocumentFragment.prototype.createAttribute = function(name) {
+    return new Node(Node.ATTRIBUTE_NODE, name, "");
```

Comment:
With a binding to `""` in absence of a value in Node, we can use `return new Node(Node.ATTRIBUTE_NODE, name)` here